frame/executive/src/lib.rs

```diff
@@ -328,14 +335,15 @@ where
 
 			Self::initialize_block(block.header());
 
-			// any initial checks
-			Self::initial_checks(&block);
+			// any initial checks, and retrieve the index of the first signed extrinsic from the
+			// `extrinsics` vec if one exists.
+			let maybe_first_non_inherent_index = Self::initial_checks(&block);
```

Comment:
Ok, so `initial_checks` gets used here... meaning block execution can panic. What would happen when `execute_block` panics?

Reply:
the client should handle it, if execute block panics, then it is invalid